frontend/components/modal.tsx

```diff
@@ -0,0 +1,85 @@
+"use client";
+
+import Link from "next/link";
+import { usePathname, useSearchParams, useRouter } from "next/navigation";
+import React, { useEffect, useRef } from "react";
+import { X } from "react-feather";
+
+interface Props {
+  modalId: number | null;
+  content: React.ReactNode;
+  _footer?: React.ReactNode;
+  okLink?: React.ReactNode;
+  openModal: React.ReactNode;
+  //onOK needs "use server"
+  _onOk?: () => Promise<void>;
+}
+
+const Modal = ({ content, _footer, _onOk, okLink, openModal, modalId }: Props) => {
+  const router = useRouter();
+  const searchParams = useSearchParams();
+  const dialogRef = useRef<null | HTMLDialogElement>(null);
+  const pathName = usePathname();
+  let showModal = searchParams.get("showModal");
+  const link = `?showModal=${modalId}`;
+
+  const dialog: JSX.Element | null =
+    showModal == modalId ? (
+      <dialog ref={dialogRef}>
+        <div
+          onMouseDown={() => closeModal()}
+          className="modal-background"
+        >
+          <div
+            onMouseDown={e => e.stopPropagation()}
+            className="modal-box"
+          >
+            <div className="modal-title">
+              <X onClick={() => closeModal()} />
```

Comment:
X voisi olla buttonin sisällä, että siihen voi tarkentaa myös tabilla. Nyt jos käyttäjä tabaa vahingossa esim. selaimen asetuksiin, ei tämä pysty palaamaan modaaliin jos siellä ei ole ainuttakana linkkiä tai buttonia. Esim näin:

`<div className="modal-title">
              <button
                onClick={() => closeModal()}
                className="button-transparent"
              >
                <X color="#eff2f2" />
              </button>
            </div>`